packages/l10n/src/create-l10n-injector/create-l10n-injector.tsx

```diff
@@ -2,19 +2,21 @@ import React from 'react';
 import { reportErrorToSentry } from '@commercetools-frontend/sentry';
 import { getDisplayName } from '../utils';
 
-type LoadLocale<LoadedData> = (locale: string) => Promise<LoadedData>;
-type InjectorOptions<LoadedData> = {
+type LoadLocale<LoadedData extends {}> = (
+  locale: string
+) => Promise<LoadedData>;
+type InjectorOptions<LoadedData extends {}> = {
   displayName: string;
   propKey: string;
   propLoadingKey: string;
   loadLocale: LoadLocale<LoadedData>;
 };
-type State<Data> = {
+type State<Data extends {}> = {
   isLoading: boolean;
-  data?: Data;
+  data?: Data | {};
```

Comment:
This is also unrelated to the issue with `act` but it's also a fix to be backwards compatible.